papermill/api.py

```diff
@@ -221,24 +221,54 @@ def display_output(self, name):
         if name not in outputs:
             raise PapermillException(
                 "Output Name '%s' is not available in this notebook.")
-        output = outputs[name]
-        ip_display(output.data, metadata=output.metadata, raw=True)
+
+        # Loop over elements of output
+        for output in outputs[name]:
+            ip_display(output.data, metadata=output.metadata, raw=True)
 
 
 def _get_papermill_metadata(nb, name, default=None):
     """Returns a dictionary of a notebook's metadata."""
     return nb.metadata.get('papermill', {}).get(name, default)
 
 
+def _cell_has_output_tag(cell):
+
+    for output in cell.get('outputs', []):
+        if 'papermill' in output.get('metadata', {}) and output.metadata.papermill.get('name'):
+            return True
+
+    return False
+
+
+def _get_cell_output_tag(cell):
+
+    if _cell_has_output_tag(cell):
+
+        names = set([output.metadata.papermill.get('name') for output in cell.outputs if 'papermill' in output.get('metadata', [])])
```

Comment:
Note, if you return an iterable of booleans you can directly iterate over the enumerated iterable to give you the conditional you need for the whole set (avoiding the redundant metadata checking

```python
names = set(cell.outputs[idx].metadata.papermill.get('name') 
            for idx, has_name in enumerate(_cell_outputs_with_names(cell)) if has_name)
```

also you don't need to wrap it in a list since you can pass the generator directly to the set in its surrounding parens.